src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java

```
@@ -36,14 +40,43 @@ public Film create(@Valid @RequestBody Film film) {
     public Film putFilm(@RequestBody Film film) {
         log.info("Получен PUT запрос к эндпоинту: /films, Строка параметров запроса: '{}'",
                 film);
+        if (filmService.getFilm(film.getId()).isEmpty()) {
+            throw new ExceptionNotFound("Не найден Film c id " + film.getId());
```

существует ещё один интересный принцип "fail fast", по котором ошибку лучше кидать сразу же, как стало понятно, что что-то пошло не так
https://russianblogs.com/article/6148454582/
вот здесь уже очень поздно, кмк, можно было намного раньше определить, что энтити не существует